From 0b4e6cdd67a57f976f0f7628a99bcb51c7de9d47 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Zolt=C3=A1n=20B=C3=B6sz=C3=B6rm=C3=A9nyi?=
 <zboszor@gmail.com>
Date: Sat, 15 Nov 2025 10:11:46 +0100
Subject: [PATCH 1/2] Use find_package(Python) for nanobind
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

nanobind explicitly needs find_package(Python),
so add that in the cross-compiling branch.

Also use Python_ARTIFACTS_PREFIX, a cmake 4.0/4.1+ feature to
separate host and target Python artifacts.

The build warns about library shadowing in recipe-sysroot-native
vs recipe-sysroot, regardless of setting CMP0190 to NEW or OLD.

Signed-off-by: Zoltán Böszörményi <zboszor@gmail.com>
Upstream-Status: Inappropriate [oe specific]
---
 CMakeLists.txt | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4a7df7d..5a43889 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -139,9 +139,15 @@ if(CMAKE_CROSSCOMPILING)
   # https://gitlab.kitware.com/cmake/cmake/-/issues/25145
   if(ONNX_BUILD_PYTHON)
     find_package(Python3 REQUIRED COMPONENTS ${python_dev_component})
+    # Find Python for nanobind
+    find_package(Python REQUIRED COMPONENTS ${python_dev_component})
   endif()
+  # Works with CMake 4.0+
+  set(Python_ARTIFACTS_PREFIX "_HOST")
   find_package(Python REQUIRED COMPONENTS Interpreter)
-  set(ONNX_PYTHON_INTERPRETER Python::Interpreter)
+  # Find Python for nanobind
+  set(Python_EXECUTABLE ${Python_HOST_EXECUTABLE})
+  set(ONNX_PYTHON_INTERPRETER ${Python_HOST_EXECUTABLE})
 else()
   find_package(Python3 REQUIRED COMPONENTS Interpreter ${python_dev_component})
   # Find Python for nanobind
-- 
2.51.1

